core/consensus/aura/src/lib.rs

```diff
@@ -391,9 +401,10 @@ impl<B: Block, C, E, I, Error, SO> SlotWorker<B> for AuraWorker<C, E, I, SO> whe
 /// if it's successful, returns the pre-header, the slot number, and the signat.
 //
 // FIXME #1018 needs misbehavior types
-fn check_header<B: Block>(slot_now: u64, mut header: B::Header, hash: B::Hash, authorities: &[AuthorityId])
-	-> Result<CheckedHeader<B::Header, ed25519::Signature>, String>
-	where DigestItemFor<B>: CompatibleDigestItem
+fn check_header<B: Block, P: Pair>(slot_now: u64, mut header: B::Header, hash: B::Hash, authorities: &[AuthorityId<P>])
+	-> Result<CheckedHeader<B::Header, P::Signature>, String>
+	where DigestItemFor<B>: CompatibleDigestItem<P>,
+		P::Public: Clone + AsRef<P::Public>,
```

Comment:
Why `AsRef<P::Public>`? Can't it just be `&_`?

Reply:
I get compiler errors otherwise.  There _should_ be a blanket impl `impl AsRef<T> for T`, but there isn’t.